pkg/cloud/services/gc/loadbalancer.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gc
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/service/elb"
+	"github.com/aws/aws-sdk-go/service/elbv2"
+	rgapi "github.com/aws/aws-sdk-go/service/resourcegroupstaggingapi"
+)
+
+func (s *Service) deleteElasticLoadbalancingResources(ctx context.Context, resources []*rgapi.ResourceTagMapping) error {
+	s.scope.V(2).Info("Deleting load balancers")
+	if err := s.deleteLoadBalancers(ctx, resources); err != nil {
+		return fmt.Errorf("deleting load balancers: %w", err)
+	}
+	if err := s.deleteTargetGroups(ctx, resources); err != nil {
+		return fmt.Errorf("deleting target groups: %w", err)
```

Comment:
Do we need to confirm that the resources are gone, like we do [when we delete the API server ELB](https://github.com/kubernetes-sigs/cluster-api-provider-aws//blob/4d3a8c63fab99a3aa9c029394bce08cf343b9855/pkg/cloud/services/elb/loadbalancer.go#L176-L182)?

Reply:
I'm undecided. This will mean checking for every resource we deleted and so will be a bunch more api calls. In e2e testing i haven't seen any issues.

Reply:
Good point.

The only reason to confirm a resource is gone before proceeding is if subsequent operations require the resource to be gone.

Let's say we do not confirm some LB is gone. We return from this function, and eventually finish garbage collection. Then, when CAPA begins deleting other resources, it may hit an unexpected DependencyViolation error. We expect this error to go away when the LB is eventually, but CAPA may have to requeue before that happens.

Reply:
The `ReconcileDelete` of the gc service can be called many times, so if we do go around the reconciliation look again for delete it won't be a problem.

Reply:
Keep in mind that some things are deleted asynchronously and you usually have to pass in something like `Wait: true` in the options so it doesn't just mark it for deletion and then proceed. We want the call to wait I think, right?

Reply:
Agreed things can be deleted asynchronously. And it would be better to `Wait` after every delete. But if we wait, then it's at least 1 extra api call per resource we delete. We have struggled with the number of API calls and throttling in the past.

As it stands,  if we delete and continue with the delete reconciliation and something like a LB isn't deleted yet then we will get a `DependencyViolation` error when deleting the VPC. The reconcile delete will be retried and it will pass. 

@Skarlso @dlipovetsky - if you both strongly feel we should be waiting then i can add this is no problem.

Reply:
Nah, I'm fine honestly with not adding a bunch of API calls. My remark was more in the lines of "let's not forget this and revisit later". :)